pandas/core/reshape.py

```diff
@@ -1153,6 +1182,44 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     from pandas.tools.concat import concat
     from itertools import cycle
 
+    # Deprecate filling NaN values with zeros, GH15926
+    # When this is finally deprecated, simply remove this block
+    # of code and change the default to np.nan in the function signature
+    # of `get_dummies`.
+    if fill_value is None:
+        warnings.warn('The default behavior of filling NaN values '
+                      'with zeros will be deprecated. Use '
+                      '`df = pd.get_dummies(df, fill_value=0)` to reproduce '
+                      'this behavior', FutureWarning, 3)
+        fill_value = 0
+
+    # Infer the proper output dtype.
+    # GH15926
+    vals = data.values.ravel() if hasattr(data, 'values') else data
```

Comment:
using ``.values`` is pretty much a no-no as if mixed dtypes this is completely non-performant. The issue is that we should *not* have multiple dtypes. this needs more work on this type of checking.